pycbc/events/status.py

```diff
@@ -0,0 +1,252 @@
+# Copyright (C) 2018 Alex Nitz
+
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 3 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+#
+# =============================================================================
+#
+#                                   Preamble
+#
+# =============================================================================
+#
+""" Utitlities to query the status of gravitational-wave instruments
+from public sources as well as dqsegdb.
+"""
+
+from glue.segments import segmentlist, segment
+import json, urllib
+
+def parse_veto_definer(veto_def_filename):
+    """ Parse a veto definer file from the filename and return a dictionary 
+    indexed by ifo and veto definer category level.
+    
+    Parameters
+    ----------
+    veto_def_filename: str
+        The path to the veto definer file
+    
+    Returns:
+        parsed_definition: dict   
+            Returns a dictionary first indexed by ifo, then category level, and
+            finally a list of veto definitions.
+    """
+    from glue.ligolw import ligolw, table, lsctables, utils as ligolw_utils
+    from glue.ligolw.ligolw import LIGOLWContentHandler as h
+    lsctables.use_in(h)
+    
+    indoc = ligolw_utils.load_filename(veto_def_filename, False, contenthandler=h)
+    veto_table = table.get_table(indoc, 'veto_definer')
+    
+    ifo = veto_table.getColumnByName('ifo')
+    name = veto_table.getColumnByName('name')
+    version = numpy.array(veto_table.getColumnByName('version'))
+    category = numpy.array(veto_table.getColumnByName('category'))
+    start = numpy.array(veto_table.getColumnByName('start_time'))
+    end = numpy.array(veto_table.getColumnByName('end_time'))
+    start_pad = numpy.array(veto_table.getColumnByName('start_pad'))
+    end_pad = numpy.array(veto_table.getColumnByName('end_pad'))
+    
+    data = {}
+    for i in range(len(veto_table)):
+        if ifo[i] not in data:
+            data[ifo[i]] = {}
+        
+        if category[i] not in data[ifo[i]]:
+            data[ifo[i]][category[i]] = []
```

Comment:
So category is just an integer so data['H1']['1'] would contain all the CAT_1 vetoes. So IIRC the macro name for `CAT_1` is currently `1`. Think this needs a `"CAT_"+` or similar here

Reply:
(And then there's the Hardware injections are still `3` problem. In the old code we remap that to "H")

Reply:
Okay, I have to make some change here because the macro name for CAT_1 is currently `int(1)` which cannot be specified as a string!

Reply:
@spxiwh How hard is that to change? It would be nice if we could get rid of this discrepency already

Reply:
Not something that's in the scope of this merge request. We can ask about this, but it requires coordination with everyone else using these, so this would have to support the current version now.